### PR TITLE
Add REST API 

### PR DESCRIPTION
**Pull request by liadomide**
REST Flack based API added.
A basic REST Client package is also created now.
ModelView for the GUI forms is introduced
Few bugs were fixes

### CODE REVIEW

File: framework_tvb/tvb/adapters/visualizers/pearson_cross_correlation.py [hunk 1]
```diff
@@ -101,7 +114,7 @@ def launch(self, datatype):
         pars = dict(matrix_labels=json.dumps([labels, labels]),
                     matrix_shape=json.dumps(matrix_shape),
                     viewer_title='Cross Corelation Matrix plot',
-                    url_base=URLGenerator.build_h5_url(datatype.gid, 'get_correlation_data', parameter=''),
+                    url_base=URLGenerator.build_h5_url(view_model.datatype, 'get_correlation_data', parameter=''),
```
[the remark this answers is not in the document]
Comment (popaula937) [reply framework_tvb/tvb/adapters/visualizers/pearson_cross_correlation.py]: yes, it should

File: framework_tvb/tvb/adapters/visualizers/fourier_spectrum.py [hunk 2]
```diff
@@ -81,25 +93,31 @@ class FourierSpectrumDisplay(ABCDisplayer):
...
 
-    def get_required_memory_size(self, **kwargs):
+    def get_required_memory_size(self, view_model):
+        # type: (FourierSpectrumModel) -> dict
         """
         Return the required memory to run this algorithm.
         """
-        return numpy.prod(kwargs['input_data'].read_data_shape()) * 8
+        fs_input_index = self.load_entity_by_gid(view_model.input_data.hex)
+        return numpy.prod(fs_input_index.read_data_shape()) * 8
```
[the remark this answers is not in the document]
Comment (popaula937) [reply framework_tvb/tvb/adapters/visualizers/fourier_spectrum.py]: indeed, it should be get_data_shape()

File: framework_tvb/tvb/tests/framework/adapters/uploaders/nifti_importer_test.py [hunk 3]
```diff
@@ -112,7 +119,7 @@ def test_import_demo_ts(self):
         """
         This method tests import of a NIFTI file.
         """
-        time_series_index = self._import(self.TIMESERIES_NII_FILE, TimeSeriesVolumeIndex)
+        time_series_index = self._import(self.TIMESERIES_NII_FILE, TimeSeriesVolumeIndex, self.connectivity_gid)
```
Comment (liadomide) [inline framework_tvb/tvb/tests/framework/adapters/uploaders/nifti_importer_test.py]: connectivity reference is not necessary in most cases of NII import.
The test was wrongly "corrected" before after a broken code in importer

File: framework_tvb/tvb/interfaces/rest/server/run.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+
+TvbProfile.set_profile(TvbProfile.COMMAND_PROFILE)
+
+LOGGER = get_logger('tvb.interfaces.rest.server.run')
+LOGGER.info("TVB application will be running using encoding: " + sys.getdefaultencoding())
+
+FLASK_PORT = 9090
+
+os.environ['FLASK_ENV'] = 'development'
```
Comment (liadomide) [inline framework_tvb/tvb/interfaces/rest/server/run.py]: Why not have this in the if is_deployment part ?

File: framework_tvb/tvb/core/adapters/abcuploader.py [hunk 5]
```diff
@@ -39,15 +39,15 @@
...
+from tvb.core.neotraits.uploader_view_model import UploaderViewModel
 
 
 class ABCUploaderForm(ABCAdapterForm):
 
     def __init__(self, prefix='', project_id=None):
         super(ABCUploaderForm, self).__init__(prefix, project_id)
-        self.subject_field = SimpleStrField(self, name=DataTypeMetaData.KEY_SUBJECT, required=True, label='Subject',
-                                            default=DataTypeMetaData.DEFAULT_SUBJECT)
+        self.subject_field = StrField(UploaderViewModel.data_subject, self, name='Data_Subject')
```
[the remark this answers is not in the document]
Comment (liadomide) [reply framework_tvb/tvb/core/adapters/abcuploader.py]: In fact, not, because the default exists on UploaderViewModel.data_subject field

**Comment by maedoc** [comment]
the PR is big.. is there a spec or doc on this API somewhere? Jan and I were wondering how this could be reused in HBP

**Comment by liadomide** [comment]
> the PR is big.. is there a spec or doc on this API somewhere? Jan and I were wondering how this could be reused in HBP

We have many tasks on Jira, if you search for REST, and we have this document (from The Virtual Brain Cloud) where we documented the API. If you want, we can also have a call, and we try to answer your questions/concerns.

https://docs.google.com/document/d/1CjEGYPVT29U3y7xSGHgsie74vVpth-NbOQu6up6886g/edit

**Pull request merged**